portfolio/src/main/webapp/script.js

```diff
@@ -47,4 +47,18 @@ function randomPic() {
   picElem.src = pics[randomFromArray(pics)]
   //make image visible
   picElem.style.display = "block"
+}
+
+/**
+ * Do hello is a dynamic approach to the /hello servlet
+ */
+async function doHello() {
+    //gets text from servlet
+    const response = await fetch("/hello")
+    const respText = await response.text()
+    console.log("Success: HelloWorldServlet response obtained")
+
+    //gets <p> to insert text
+    const respDisplayElem = document.getElementById("dynamicHello")
+    respDisplayElem.innerText = respText
```

Comment:
Javascript code will work without semicolons at the end of the line, since the parser adds it. But it is a good idea to add them yourself. I didn't notice this earlier.